Scripts/Python/xJournalBookGUIPopup.py

```diff
@@ -79,6 +82,14 @@
 # globals
 LocalAvatar = None
 
+# pfJournalBook's EsHTML will only allow links with integer events.
+# This isn't very friendly for age creators, so we'll let them specify
+# a url by doing <a href="https://foo.com">, and this regex will
+# convert that to an integer id. It will do the same for any tag, but
+# only <img> and <movie> are useful.
+_URL_REGEX = re.compile(r"(?P<href>href\s*=\s*(?P<quote>[\"\']?)\s*?(?P<url>https?:\/\/[^\s>]+)\s*?(?P=quote))(?=[^<]*>)", re.IGNORECASE)
+_HOST_REGEX = re.compile(R"(?:https?:\/\/)?(?:www\.)?([^\/?#]+).*", re.IGNORECASE)
```

Comment:
Would be really cool if we could make `urllib.parse` usable inside Plasma at some point...

Reply:
That would be some good future work. It looks like pulling in just `urllib.parse` would be safe. It's `urllib.request` that we especially don't want due to the dependency on `socket`. `urllib.parse` doesn't seem to depend on any other `urllib` submodules.